### ISSUE TITLE
count(): Parameter must be an array or an object that implements Countable

### ISSUE THREAD

**Issue by ahmed-sigmalux** [cloudflare/Cloudflare-WordPress#210]
After upgrading to PHP 7.2, I'm seeing this repeatedly in my debug.log:

`PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /var/www/wordpress/plugins/cloudflare/src/WordPress/Hooks.php on line 133`

**Comment by ahmed-sigmalux** [comment]
Has anyone at Cloudflare seen this bug?

**Comment by piscis** [comment]
This is what we get since the upgrade from php 7.0 to 7.2:
```
PHP Warning: count(): Parameter must be an array or an object that implements Countable in .... /plugins/cloudflare/src/WordPress/Hooks.php on line 133
```

**Comment by michalhalas** [comment]
Can confirm I've been receiving same error with newest available version of plugin 3.3.2 and php version 7.0.25
`Warning: count(): Parameter must be an array or an object that implements Countable in /var/www/html/wp-content/plugins/cloudflare/src/WordPress/Hooks.php on line 133`

**Comment by janvitos** [comment]
Same thing here. I noticed the issue in the NGINX error log after upgrading to PHP 7.2.

```
[error] 1140#1140: *369161 FastCGI sent in stderr: "PHP message: PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /wordpress/wp-content/plugins/cloudflare/src/WordPress/Hooks.php on line 133" while reading response header from upstream, client: xxx.xxx.xxx.xxx, server: www.com, request: "POST /wordpress/wp-admin/admin-ajax.php HTTP/1.1", upstream: "fastcgi://unix:/var/run/php-fpm/php-fpm.sock:", host: "www.com", referrer: "/wordpress/wp-admin/post-new.php"
```

**Comment by ron2122** [comment]
same error here


Warning: count(): Parameter must be an array or an object that implements Countable in /home/blocktoken/public_html/wp-content/plugins/cloudflare/src/WordPress/Hooks.php on line 133

Warning: Cannot modify header information - headers already sent by (output started at /home/blocktoken/public_html/wp-content/plugins/cloudflare/src/WordPress/Hooks.php:133) in /home/blocktoken/public_html/wp-includes/pluggable.php on line 1216

**Comment by conorbarclay** [comment]
Still getting count() errors on PHP 7.2 here. Latest version of plugin.

**Comment by janvitos** [comment]
I'm pretty sure the bug stems from the fact that the count() function is used to count arrays and NOT integers. But in Hooks.php, it is actually trying to count an integer, here's why.

On line 133 from Hooks.php where the warning is triggered, you have the following line of code:
```
if (count($wpDomain) <= 0) {
```
$wpDomain is actually NOT an array and is declared as a value from an array on line 132:
```
$wpDomain = $wpDomainList[0];
```
Replacing line 133 with the following code fixes the warning:
```
if (count($wpDomainList) <= 0) {
```
I will do a pull request for this.

**Comment by awbauer** [comment]
This also breaks Gutenberg for end users, showing "Updating failed" due to the broken JSON response. Deactivating the plugin until a patch is pushed.

<img width="998" alt="screenshot 2018-09-07 17 25 13" src="https://user-images.githubusercontent.com/1152149/45243743-ff371680-b2c2-11e8-80ba-6ef639621eb7.png">

**Comment by thinknirmal** [comment]
Been nine months and no fix? At least please merge the patch someone submitted #221

**Comment by pentago** [comment]
Any progress on this one yet? 
Thanks!

**Comment by AiratHalitov** [comment]
Fixed in #231

**Comment by stanwmusic** [comment]
Thanks!
